openforcefield/tests/test_toolkits.py

```diff
@@ -236,6 +236,35 @@ def test_from_openeye_implicit_hydrogen(self):
         molecule_from_expl = Molecule.from_openeye(oemol_expl)
         assert molecule_from_expl.to_smiles() == molecule_from_impl.to_smiles()
 
+    @pytest.mark.skipif(not OpenEyeToolkitWrapper.is_available(), reason='OpenEye Toolkit not available')
+    def test_openeye_from_smiles_hydrogens_are_explicit(self):
+        """
+        Test to ensure that OpenEyeToolkitWrapper.from_smiles has the proper behavior with
+        respect to its hydrogens_are_explicit kwarg
+        """
+        toolkit_wrapper = OpenEyeToolkitWrapper()
+        smiles_impl = "C#C"
+        with pytest.raises(ValueError,
+                           match="but OpenEye Toolkit interpreted SMILES 'C#C' as having implicit hydrogen") as excinfo:
+            offmol = Molecule.from_smiles(smiles_impl,
+                                          toolkit_registry=toolkit_wrapper,
+                                          hydrogens_are_explicit=True)
+        offmol = Molecule.from_smiles(smiles_impl,
+                                      toolkit_registry=toolkit_wrapper,
+                                      hydrogens_are_explicit=False)
```

Comment:
I'd suggest adding an assert statement here simply checking that the number of atoms is the expected one.

Reply:
👍 Done.